### PR TITLE
Fix commits page url

### LINKED ISSUE #2805: Recent "Url scheme" commit breaks redirects (404s) when the "root url" is in a subdirectory
- Gitea version (or commit ref): https://github.com/go-gitea/gitea/commit/513375c429435ba60a667b219bdfb00e5b760b38
- Git version: 2.14.3
- Operating system: Linux
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [X] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [X] Not relevant
- Log gist:

## Description

Use a `ROOT_URL` in your config in a directory other than the "root" directory. For example, https://example.com/git/

The problem is that the "Commits" link at the top of a repository view has the form `/BASE_DIR/COMPANY_NAME/REPOSITORY_NAME/commits/master` when it should be `/BASE_DIR/COMPANY_NAME/REPOSITORY_NAME/commits/branch/master` (as per the latest commit: https://github.com/go-gitea/gitea/commit/513375c429435ba60a667b219bdfb00e5b760b38 ).

However, if you click the current "Commits" link it incorrect drops the "subdirectory" part of the URL. Meaning Gitea currently incorrectly redirects to `/COMPANY_NAME/REPOSITORY_NAME/commits/branch/master` instead of `/BASE_DIR/COMPANY_NAME/REPOSITORY_NAME/commits/branch/master`. Meaning, the current redirection code assumes the gitea installation is in the root directory. Which, for us that's not true.

The breaking changes were in this commit: https://github.com/go-gitea/gitea/commit/513375c429435ba60a667b219bdfb00e5b760b38

### PR DESCRIPTION
**Pull request by Morlinest**
Fixes #2805.

### CODE REVIEW

File: templates/repo/sub_menu.tmpl [hunk 1]
```diff
@@ -2,7 +2,7 @@
 	<div class="ui two horizontal center link list">
 		{{if and (.Repository.UnitEnabled $.UnitTypeCode) (not .IsBareRepo)}}
 			<div class="item{{if .PageIsCommits}} active{{end}}">
-				<a href="{{.RepoLink}}/commits/{{EscapePound .BranchName}}"><i class="octicon octicon-history"></i> <b>{{.CommitsCount}}</b> {{.i18n.Tr "repo.commits"}}</a>
+				<a href="{{.RepoLink}}/commits/branch/{{EscapePound .BranchName}}"><i class="octicon octicon-history"></i> <b>{{.CommitsCount}}</b> {{.i18n.Tr "repo.commits"}}</a>
```
Comment (lafriks) [inline templates/repo/sub_menu.tmpl]: Shouldn't there needs to be url depending on what page you are browsing (tags/commits/branches)?

Comment (Morlinest) [reply templates/repo/sub_menu.tmpl]: You are right, I have to do check for `tags`.

**Comment by Morlinest** [comment]
@lafriks Done.

**Comment by ethantkoenig** [comment]
LGTM

**Comment by lafriks** [comment]
LGTM

**Pull request merged**